pilot/pkg/proxy/envoy/v2/pushqueue_test.go

```diff
@@ -182,35 +183,48 @@ func TestProxyQueue(t *testing.T) {
 
 	t.Run("concurrent", func(t *testing.T) {
 		p := NewPushQueue()
+		key := func(p *XdsConnection, eds string) string { return fmt.Sprintf("%s~%s", p.ConID, eds) }
 
-		go func() {
+		// We will trigger many pushes for eds services to each proxy. In the end we will expect
+		// all of these to be dequeue, but order is not deterministic.
+		expected := map[string]struct{}{}
+		for eds := 0; eds < 100; eds++ {
 			for _, pr := range proxies {
-				p.Enqueue(pr, &PushInformation{})
+				expected[key(pr, fmt.Sprintf("%d", eds))] = struct{}{}
 			}
-		}()
-		errs := make(chan error)
-		wg := sync.WaitGroup{}
-		wg.Add(1)
+		}
 		go func() {
-			expect := 0
-			for {
-				ExpectDequeue(t, p, proxies[expect])
-				expect++
-				if expect == len(proxies) {
-					wg.Done()
-					break
```

Comment:
This is just testing that there is nothing left in the queue after at most 10s.
Is there anything else this should be testing ?

Reply:
it's checking `expected` is empty meaning we have received all of the pushes we expect - first we populate it with what we expect and remove them as we see them